### DISCUSSION TITLE
Bonsai suddenly started crashing because of memory leakage

### DISCUSSION THREAD

**Discussion by noamhed** [bonsai-rx/bonsai#1866] [Q&A]
Hey, I am a lab assistant in a lab that uses bonsai to record experiments. Recently, after working as intended, bonsai began using 100% CPU and flooding the RAM. Does someone have a fix for it? I'm adding all the computer specs while the program is running and after it finish
![WhatsApp Image 2024-07-01 at 14 05 28 (1)](https://github.com/bonsai-rx/bonsai/assets/142654021/3b0d5669-6030-4384-99a1-763c2c502d58)
![WhatsApp Image 2024-07-01 at 14 05 28 (2)](https://github.com/bonsai-rx/bonsai/assets/142654021/46a006a8-8826-4550-a303-02ebc63e5a45)
![WhatsApp Image 2024-07-01 at 14 05 28 (3)](https://github.com/bonsai-rx/bonsai/assets/142654021/51e39ef6-749a-43ca-8123-f8dfb918c745)
![WhatsApp Image 2024-07-01 at 14 05 28](https://github.com/bonsai-rx/bonsai/assets/142654021/104dc9cf-8130-46f3-8bcb-71e567437a21)
![WhatsApp Image 2024-07-01 at 14 05 29](https://github.com/bonsai-rx/bonsai/assets/142654021/746819fa-4d5c-4bff-9299-1a0b92dc853b)
![WhatsApp Image 2024-07-01 at 14 07 31](https://github.com/bonsai-rx/bonsai/assets/142654021/db793923-c069-4bf7-81c0-4c408c4a5abd)

**Comment by bruno-f-cruz** [comment]
I am not sure this qualifies as a memory leak. Looking at the console output, and assuming that you encoding fps is the same as your camera, it seems like your machine cant keep up with the encoding process real time (hence the 0.3x output).  Since it cant keep up, it keeps accumulating the frames in memory until they can be handled. Change the codec to a more efficient one and make sure that, if your codec targets the GPU, you are targeting the dedicated one.

**Reply by noamhed** [reply]
Thank you so much for the quick response! 
I changed the graphics settings to target the GPU through Windows setting, the CPU isn't maxed out now but there is still a RAM overflow problem. 
I'm sorry for my ignorance but I've been trying for hours to figure out how to change the codec and I just can't figure it out. can you please refer me to a guide or explain what are the steps required to change the VideoWriter codec? 
thanks again, Noam
